File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        d = {}
        for c in s:
            d[c] = d.get(c,0)+1
        for c in range(len(s)):
            if d[s[c]] == 1:
                return c
        return -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        Dictionary<char,int> d=new();

        foreach(var c in s){
            if(!d.TryAdd(c,1)){
                d[c]++;
            }
        }
        foreach(var c in s){
            if(d[c]==1){
                return s.IndexOf(c);
            }
        }
        return -1;
    }
}